### PR TITLE
Fix DPA reconcile on empty BSL secret name and key

### PR DESCRIPTION
**Pull request by sbahar619**
This PR addresses: 
https://issues.redhat.com/browse/OADP-2940 

I added coverage for fail DPA reconcile on empty BSL credential name and key.
Also, I fixed the logic condition which was not accurate.
I added 4 more unit tests for more coverage.

### CODE REVIEW

**Review by kaovilai — DISMISSED** [review]

File: controllers/validator.go [hunk 1]
```diff
@@ -54,7 +54,21 @@ func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error)
...
 		// Check the Velero flags 'no-secret' or 'no-default-backup-location' are not set
-		if !dpa.Spec.Configuration.Velero.HasFeatureFlag("no-secret") || dpa.Spec.Configuration.Velero.NoDefaultBackupLocation {
+		if !(dpa.Spec.Configuration.Velero.HasFeatureFlag("no-secret") || dpa.Spec.Configuration.Velero.NoDefaultBackupLocation) {
+
+			// Check if the user specified credential under velero
+			if location.Velero != nil && location.Velero.Credential != nil {
+
+				// Check if user specified empty credential key
+				if location.Velero.Credential.Key == "" {
+					return false, fmt.Errorf("Secret key specified in BackupLocation name %s cannot be empty", location.Name)
```
Comment (shubham-pampattiwar) [inline controllers/validator.go:64]: NIT: remove `name` string from log

File: controllers/validator.go [hunk 2]
```diff
@@ -54,7 +54,21 @@ func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error)
...
+				}
+
+				// Check if user specified empty credential name
+				if location.Velero.Credential.Name == "" {
+					return false, fmt.Errorf("Secret name specified in BackupLocation name %s cannot be empty", location.Name)
```
Comment (shubham-pampattiwar) [inline controllers/validator.go:69]: NIT: remove `name` string from log

Comment (sbahar619) [reply controllers/validator.go]: Thanks, done.

File: controllers/validator.go (hunk 1 above)
Comment (sbahar619) [reply controllers/validator.go]: Thanks, done.

**Comment by weshayutin** [comment]
/hold  
this issue has been kicked out to 1.3.2

**Comment by kaovilai** [comment]
to add to this PR: https://github.com/openshift/oadp-operator/pull/1220#pullrequestreview-1719036142 for 1.3.2

**Review by kaovilai — APPROVED** [review]

**Review by sseago — APPROVED** [review]

**Comment by weshayutin** [comment]
anybody know what ` Context retired without replacement.` means? Looking around don't see any reference

**Comment by kaovilai** [comment]
> anybody know what  Context retired without replacement. means? Looking around don't see any reference

See [here](https://github.com/k8s-ci-robot/test-infra/blob/d7867ec05b41cd5d0b69d53e52811a5c9a8d5169/maintenance/migratestatus/migrator/migrator.go#L140)

**Pull request merged**